### PR TITLE
Add new Devops agent for Python

### LINKED ISSUE #28: Request to add more pod template for newer DevOps agents
Currently, we've add some fresh DevOps agents in https://github.com/kubesphere/devops-agent, please see:

- Add new DevOps agent for Golang 1.16.8 ([#39](https://github.com/kubesphere/devops-agent/pull/39))
- Add new DevOps agent for Maven with JDK 11 ([#38](https://github.com/kubesphere/devops-agent/pull/38/files))
- Add new DevOps agent for Python 3.7 ([#31](https://github.com/kubesphere/devops-agent/pull/31))
- Add new DevOps agent for Dotnet 5 ([#24](https://github.com/kubesphere/devops-agent/pull/24))

It's time to add them into `jenkins-casc-config` ConfigMap.

/kind feature

### PR DESCRIPTION
**Pull request by mangoGoForward**
### What type of PR is this?

/kind feature

### What this PR does / why we need it:
Currently, we've add some fresh DevOps agents in https://github.com/kubesphere/devops-agent, The PR try to add remained config to jenkins-casc-config

### Which issue(s) this PR fixes:

Fixes #28

### CODE REVIEW

**Comment by mangoGoForward** [comment]
Could you help me to review this PR and get me some suggestions? @JohnNiang

**Comment by LinuxSuRen** [comment]
> /kind hacktoberfest-accepted

This kind of action should be taken by the maintainer. Any way, thanks for your reminder.

**Comment by mangoGoForward** [comment]
> > /kind hacktoberfest-accepted
> 
> This kind of action should be taken by the maintainer.
> 
> I'll test it later.
> 
> Anyway, thanks for your reminder.

Haha, It's my misunderstanding about the activity's description. Thanks a lot.

File: charts/ks-devops/charts/jenkins/values.yaml [hunk 1]
```diff
@@ -430,6 +430,12 @@ Agent:
     golang16:
       image: builder-go
       tag: v3.2.0-1.16
+    Python:
+      Image: builder-python
+      Tag: v3.1.0
+    Dotnet:
+      Image: builder-dotnet
+      Tag: v3.1.0
```
Comment (JohnNiang) [inline charts/ks-devops/charts/jenkins/values.yaml:438]: Currently, [docker image](https://hub.docker.com/r/kubespheredev/builder-dotnet/tags) of `builder-dotnet` has gone. When it is getting ready, we'll merge this PR for further test.

Comment (mangoGoForward) [reply charts/ks-devops/charts/jenkins/values.yaml:438]: How about remove `builder-dotnet`, only support `builder-python` in this PR, and create a new PR for `builder-dotnet`?

Comment (JohnNiang) [reply charts/ks-devops/charts/jenkins/values.yaml:438]: That would be better.

Comment (mangoGoForward) [reply charts/ks-devops/charts/jenkins/values.yaml]: Done.

**Review by LinuxSuRen — COMMENTED** [review]
I have tested it manually. It works well. See the Jenkinsfile below:
```groovy
pipeline {
    agent {
        label 'python'
    }
    
    stages {
        stage('test') {
            steps {
                sh '''
                python -V
                '''
            }
        }
    }
}
```

Test output:
```shell
Running on python-t742f in /home/jenkins/agent/workspace/test
[Pipeline] {
[Pipeline] stage
[Pipeline] { (test)
[Pipeline] sh
+ python -V
Python 2.7.13
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
Finished: SUCCESS
```

/lgtm

@mangoGoForward Thanks for your contribution!

File: charts/ks-devops/charts/jenkins/templates/jenkins-casc-config.yml [hunk 2]
```diff
@@ -243,6 +243,72 @@ data:
...
+                workspaceVolume:
+                  emptyDirWorkspaceVolume:
+                    memory: false
+                volumes:
+                - hostPathVolume:
+                    hostPath: "/var/run/docker.sock"
+                    mountPath: "/var/run/docker.sock"
+                - hostPathVolume:
+                    hostPath: "/var/data/jenkins_python_cache"
+                    mountPath: "/home/jenkins/python/app"
```
[the remark this answers is not in the document]
Comment (mangoGoForward) [reply charts/ks-devops/charts/jenkins/templates/jenkins-casc-config.yml:279]: It just a habit for me, and specify the storage location of python service. It seems unnecessary in this mount path, I want to remove the suffix `/app`, make it more clear.

[the remark this answers is not in the document]
Comment (mangoGoForward) [reply charts/ks-devops/charts/jenkins/templates/jenkins-casc-config.yml:279]: Thanks your advice. But there may should specify the third-party library installed by `pip`, How about change this path to  cache those library?

[the remark this answers is not in the document]
Comment (JohnNiang) [reply charts/ks-devops/charts/jenkins/templates/jenkins-casc-config.yml:279]: I searched from GitHub cache action, and there are two references for us about Python cache: 
- https://github.com/actions/cache/blob/main/examples.md#python---pip
- https://github.com/actions/cache/blob/main/examples.md#python---pipenv

[the remark this answers is not in the document]
Comment (mangoGoForward) [reply charts/ks-devops/charts/jenkins/templates/jenkins-casc-config.yml]: Done.

**Comment by LinuxSuRen** [comment]
hi @mangoGoForward and @JohnNiang , in order to avoid unnecessary problems during the v3.2.0 release time period. I'll hold this PR until KubeSphere v3.2.0 is ready.
/hold

**Comment by JohnNiang** [comment]
/retitle Add new Devops agent for Python

**Comment by JohnNiang** [comment]
Hi @LinuxSuRen , since that [KubeSphere 3.2.0](https://github.com/kubesphere/kubesphere/releases/tag/v3.2.0) has been released and announced, we can continue to push this forward.

File: charts/ks-devops/charts/jenkins/values.yaml [hunk 3]
```diff
@@ -430,6 +430,9 @@ Agent:
     golang16:
       image: builder-go
       tag: v3.2.0-1.16
+    Python:
```
Comment (JohnNiang) [inline charts/ks-devops/charts/jenkins/values.yaml:433]: Hi @mangoGoForward , can you make those variable name camel-case? Please don't forget change its references.

Comment (mangoGoForward) [reply charts/ks-devops/charts/jenkins/values.yaml:433]: Ok.

Comment (mangoGoForward) [reply charts/ks-devops/charts/jenkins/values.yaml]: Done.

File: charts/ks-devops/charts/jenkins/values.yaml [hunk 4]
```diff
@@ -430,6 +430,9 @@ Agent:
     golang16:
       image: builder-go
       tag: v3.2.0-1.16
+    python:
+      Image: builder-python
```
Comment (JohnNiang) [inline charts/ks-devops/charts/jenkins/values.yaml:434]: Please don't forget the variable names `Image` and `Tag`. : P

**Review by JohnNiang — CHANGES_REQUESTED** [review]

**Comment by mangoGoForward** [comment]
Do we have schedule to refactor this file with caml-case code style? And when we add more pod template, the file seems will be poor readability.

**Comment by JohnNiang** [comment]
> Do we have schedule to refactor this file with caml-case code style? And when we add more pod template, the file seems will be poor readability.

Please follow up https://github.com/kubesphere-sigs/ks-devops-helm-chart/issues/26

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
hi @mangoGoForward it looks like everything is great. Before we merge it, you need to fix the conficts.

**Review by LinuxSuRen — COMMENTED** [review]
/lgtm

**Comment by mangoGoForward** [comment]
> hi @mangoGoForward it looks like everything is great. Before we merge it, you need to fix the conficts.

Solved. The git configuration actually has some errors in my local environment, thanks your reminder.

**Pull request merged**